webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +87,1137 @@ public WebSiteSessionController createComponentSessionController(
-            end = listeIcones.indexOf(',', begin);
-            while (end != -1) {
-              listIcons.add(listeIcones.substring(begin, end));
-              begin = end + 1;
-              end = listeIcones.indexOf(',', begin);
-            }
-          }
+    return
+        "/webSites/jsp/classifyDeclassify.jsp?Action=" + action + "&TopicId=" + id + "&Path=" +
+            linkedPathString;
+  }
 
-          /* recuperation de l'id */
-          id = scc.getNextId();
+  private String updateTopic(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
 
-          /* Creer le repertoire id */
-          scc.createFolder(scc.getWebSitePathById(id));
+    String id = request.getParameter("ChildId");
+    String path = request.getParameter("Path");
 
-          // Persist siteDetail inside database type 0 = site cree
-          SiteDetail descriptionSite =
-              new SiteDetail(id, scc.getComponentId(), nomSite, description, nomPage, 0, null, null,
-                  0, popup);
-          descriptionSite.setPositions(positions);
+    NodeDetail folderDetail = scc.getFolderDetail(id);
+    request.setAttribute("CurrentFolder", folderDetail);
 
-          String pubId = scc.createWebSite(descriptionSite);
-          descriptionSite = scc.getWebSite(id);
-          scc.setSessionSite(descriptionSite);
+    return "/webSites/jsp/updateTopic.jsp?ChildId=" + id + "&Path=" + path;
+  }
 
-          if (listIcons.size() > 0) {
-            scc.associateIcons(id, listIcons);
-          }
+  private String addTopic(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
 
-          currentPath = scc.getWebSitePathById(id);
+    String action = request.getParameter("Action");// =Add
+    String fatherId = request.getParameter("Id");
+    String newTopicName = request.getParameter("Name");
+    String newTopicDescription = request.getParameter("Description");
 
-          /* ajout de la page principale */
-          String code = " ";
+    NodeDetail folder =
+        new NodeDetail("X", newTopicName, newTopicDescription, null, null, null, "0", "X");
+    scc.addFolder(folder, "");
 
-          /* Creer la page principale */
-          scc.createFile(currentPath, nomPage, code);
+    return "/webSites/jsp/addTopic.jsp?Action=" + action + "&Id=" + fatherId;
+  }
 
-          /* publications : classer le site dans les themes cochés */
-          ArrayList<String> arrayToClassify = new ArrayList<String>();
-          boolean publish = false;
-          begin = 0;
-          end = 0;
-          end = listeTopics.indexOf(',', begin);
-          String idTopic;
-          while (end != -1) {
-            idTopic = listeTopics.substring(begin, end);
+  private String organizeSites(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "userRole". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1172)